src/utils.ts

```diff
@@ -0,0 +1,282 @@
+import { Dialog } from '@jupyterlab/apputils';
+import { Widget } from '@lumino/widgets';
+import { Message } from '@lumino/messaging';
+
+/**
+ * Time unit options for the dropdown
+ */
+export enum TimeUnit {
+  SECONDS = 'seconds',
+  MINUTES = 'minutes',
+  HOURS = 'hours',
+}
+
+/**
+ * Configuration options for the TimeInputDialog
+ */
+export interface ITimeInputDialogOptions {
+  title?: string;
+  label?: string;
+  placeholder?: string;
+  defaultValue?: number;
+  defaultUnit?: TimeUnit;
+  okLabel?: string;
+  cancelLabel?: string;
+  host?: HTMLElement;
+}
+
+/**
+ * Result returned by the TimeInputDialog
+ */
+export interface ITimeInputResult {
+  value: number;
+  unit: TimeUnit;
+  totalSeconds: number;
+}
+
+/**
+ * Custom widget for time input with numeric field and unit dropdown
+ */
+class TimeInputWidget extends Widget {
+  private _input: HTMLInputElement;
+  private _select: HTMLSelectElement;
+  private _defaultValue: number | null;
+  private _defaultUnit: TimeUnit;
+
+  constructor(options: ITimeInputDialogOptions = {}) {
+    super({ node: TimeInputWidget.createNode(options) });
+    this._defaultValue = options.defaultValue ?? null;
+    this._defaultUnit = options.defaultUnit ?? TimeUnit.SECONDS;
+
+    this._input = this.node.querySelector(
+      '.jp-notify-time-input-field',
+    ) as HTMLInputElement;
+    this._select = this.node.querySelector(
+      '.jp-notify-time-unit-select',
+    ) as HTMLSelectElement;
+
+    this._setupInputs();
+    this._setupEventListeners();
+  }
+
+  /**
+   * Create the DOM node for the widget
+   */
+  private static createNode(options: ITimeInputDialogOptions): HTMLElement {
+    const node = document.createElement('div');
+    node.className = 'jp-notify-time-input-dialog-body';
+
+    const label =
+      options.label || 'Default: seconds, +m for minutes, +h for hours:';
+
+    node.innerHTML = `
+      <label class="jp-notify-time-input-label">${label}</label>
+      <div class="jp-notify-time-input-container">
+        <input
+          type="number"
+          class="jp-notify-time-input-field"
+          placeholder="${options.placeholder || '30'}"
+          min="0"
+          step="any"
+        />
+        <select class="jp-notify-time-unit-select">
+          <option value="${TimeUnit.SECONDS}">Seconds (s)</option>
+          <option value="${TimeUnit.MINUTES}">Minutes (m)</option>
+          <option value="${TimeUnit.HOURS}">Hours (h)</option>
+        </select>
+      </div>
+      <div class="jp-notify-time-input-error" role="alert">Please enter a valid positive number</div>
+    `;
```

Comment:
I would be worried about a potential for unsanitised injection here, could this be a react-based widget instead? If not, let's set the placeholer and label separately by querying relevant nodes.